src/tainting/Taint.mli

```diff
@@ -19,23 +36,26 @@ type source = {
         this label may have changed, for instance by being propagated to
         a different label.
       *)
+  precondition : (taint list * precondition) option;
```

Comment:
It would better if we could just keep everything as a `Taint_set.t` instead of `taint list`s. Are duplicates desirable in the precondition?

Reply:
no, but this will require that we make `taint` mutually recursive with `Taint_set`, and i don't actually even know how you would do that, easily. recursive modules? it seems like a big hassle, what do you think?

Reply:
Ooops yeah I didn't realize they would become recursive. OK let's deal with this later on during a clean up. But yeah the most obvious solution would be using recursive modules.